llmfoundry/callbacks/hf_checkpointer.py

```diff
@@ -185,8 +189,16 @@ def __init__(
         self.flatten_imports = flatten_imports
         self.using_peft = False
 
+        self.final_register_only = final_register_only
+
         # mlflow config setup
         self.mlflow_registered_model_name = mlflow_registered_model_name
+        if self.final_register_only and self.mlflow_registered_model_name is None:
```

Comment:
i think this error is a bit overkill. If I'm not registering anything, its probably okay for the code to accept either value of `final_register_only`. Not super opinionated, so feel free to leave if you disagree.

Reply:
To confirm, you're suggesting that if final_register_only is True and mlflow_registered_model_name is None, we save the final model as normal?

I think that's fair since we fallback to normal save if register fails anyway. I'll raise a warning here instead of a error.

Reply:
yeah that sounds good to me

Reply:
done